src/textual/widgets/_radio_set.py

```diff
@@ -87,62 +92,68 @@ def __init__(
             disabled=disabled,
         )
 
-    @property
-    def _buttons(self) -> DOMQuery[RadioButton]:
-        """The buttons within the set."""
-        return self.query(RadioButton)
-
     def on_mount(self) -> None:
         """Perform some processing once mounted in the DOM."""
+
+        # Get the collection of buttons we're directly responsible for. We
+        # do this in here rather than in __init__ because someone might be
+        # composing the RadioSet using it as a context manager; point being
+        # there won't be any buttons being passed to __init__.
+        self._buttons = list(self.query(RadioButton))
```

Comment:
Are we creating a limitation by calculating this on mount? What if a RadioButton is removed or added after the mount?

Reply:
Yup, this was a choice, albeit one I'm not wedded to. It raises the question of what we should do with compound widgets where we don't really want the content changed via the normal DOM-modification mechanisms (see ListView, and I think a couple of others, for different but related issues). 

Happy to return this aspect back to pure DOM querying. 

Meta question though: do you think we should have the concept of a sealed container for times like this, when it makes sense to have full control over the content?

Reply:
Cool, I'll raise an issue to this effect when I'm next at a physical keyboard.